finagle-core/src/main/scala/com/twitter/finagle/dispatch/ServerDispatcher.scala

```diff
@@ -103,8 +114,30 @@ class SerialServerDispatcher[Req, Rep](
     service.close()
   }
 
-  protected def dispatch(req: Req, eos: Promise[Unit]) =
+  override def dispatch(req: Req, eos: Promise[Unit]) =
     service(req) ensure eos.setDone()
 
-  protected def handle(rep: Rep) = trans.write(rep)
+  override def handle(rep: Rep) = trans.write(rep)
+}
+
+/**
+  * A wrapper type around a 
+  * [[com.twitter.finagle.dispatcher.ServerDispather ServerDispatcher]]
+  *
+  * This type adds a 
+  * [[com.twitter.finagle.tracing.Annotation.WireRecv WireRecv]]
+  * once a dispatcher has received a request, and a
+  * [[com.twitter.finagle.tracing.Annotation.WireSend WireSend]] on a succesful
+  * write by a transporter.
+ */
+final class AnnotatedServerDispatcher[Req, Rep, In, Out, A <: ServerDispatcher[Req, Rep, In, Out]]
```

Comment:
It shouldn't be - that was leftover from a functional attempt to return the same `type` passed in, and I had just `yy` the original method definition.

Fixed.